test/uk/gov/hmrc/tai/controllers/income/IncomeControllerSpec.scala

```diff
@@ -191,15 +200,452 @@ class IncomeControllerSpec extends PlaySpec
+        .thenReturn(Future.successful(employments))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "PotentiallyCeased")(FakeRequest())
+
+      status(result) mustBe OK
+      contentAsJson(result) mustBe expectedJsonEmpty
+
+    }
+
+    "return list of live and matched pension TaxCodeIncomes for a given year" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(employments))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "PensionIncome", "Live")(FakeRequest())
+
+      status(result) mustBe OK
+
+      val expectedJson =
+        Json.parse(
+          """
+            |{
+            |    "data":[
+            |       {
+            |          "taxCodeIncome":{
+            |             "componentType":"PensionIncome",
+            |             "employmentId":1,
+            |             "amount":1100,
+            |             "description":"PensionIncome",
+            |             "taxCode":"1150L",
+            |             "name":"Employer1",
+            |             "basisOperation":"Week1Month1BasisOperation",
+            |             "status":"Live",
+            |             "inYearAdjustmentIntoCY":0,
+            |             "totalInYearAdjustment":0,
+            |             "inYearAdjustmentIntoCYPlusOne":0
+            |          },
+            |          "employment":{
+            |             "name":"company name",
+            |             "payrollNumber":"888",
+            |             "startDate":"2019-05-26",
+            |             "annualAccounts":[
+            |
+            |             ],
+            |             "taxDistrictNumber":"",
+            |             "payeNumber":"",
+            |             "sequenceNumber":1,
+            |             "cessationPay":100,
+            |             "hasPayrolledBenefit":false,
+            |             "receivingOccupationalPension":true
+            |          }
+            |       }
+            |    ],
+            |    "links":[
+            |
+            |    ]
+            | }
+          """.stripMargin)
+
+      contentAsJson(result) mustBe expectedJson
+    }
+
+    "return empty json when there are no matching live employments" in {
+      val mockIncomeService = mock[IncomeService]
```

Comment:
The test is so long and the setup is so far away it's difficult to see why there are no matching cases in this test

Reply:
The match is done using the sequence number, so if they are different, no matching live employments are found, hence the empty response... would a comment in the test explaining this suffice?